tpl/template_funcs.go

```diff
@@ -1392,6 +1392,44 @@ func replace(a, b, c interface{}) (string, error) {
 	return strings.Replace(aStr, bStr, cStr, -1), nil
 }
 
+// partialCache represents a cache of partials protected by a mutex.
+type partialCache struct {
+	mu sync.RWMutex
```

Comment:
I prefer to embed the mutex when the type is unexported and the mutex is so connected to the type, then you can do `c.Lock()` etc.